### PR TITLE
A first pass on architecture.text

### PR DESCRIPTION
**Pull request by fabianp**
Hi there!

Oldie here (contributor n. 2) although I haven't been active in a long time. Ondrej was kind enough to notify me about this initiative. 

I went through the paper and made some minor changes in architecture.tex: Some rephrasing and add a phrase at the beginning of the section introducing the content. Let me know what you think.

Also, please let me know where would my efforts be best invested in the future.

### CODE REVIEW

File: architecture.tex [hunk 1]
```diff
@@ -5,15 +5,17 @@
 
 \subsection{The Core}
 
-Every symbolic expression in SymPy is an instance of a Python class.
+The core of a computer algebra system (CAS) refers to the module who is in
```
Comment (asmeurer) [inline architecture.tex]: "who" -> "that"

File: architecture.tex [hunk 2]
```diff
@@ -177,13 +179,13 @@ \subsection{Extensibility}
 SymPy class, generally either \texttt{Basic}, \texttt{Expr}, or
 \texttt{Function}. All SymPy classes used for expression trees~\footnote{Some
   internal classes, such as those used in the polynomial module, do not follow
-  this rule.} should be subclasses of the base class
+  this rule for efficiency reasons.} should be subclasses of the base class
 \texttt{Basic}, which defines some basic methods for symbolic expression
 trees. \texttt{Expr} is the subclass for mathematical expressions that can be
-added and multiplied together. Instances of \texttt{Expr} are typically
+added and multiplied together. Instances of \texttt{Expr} are typically expressions involving
```
Comment (asmeurer) [inline architecture.tex]: That's not really what is meant here. What I mean is that the expressions represent some complex number, like `cos(x) + 1` is the complex number cos(x) + 1.

File: architecture.tex [hunk 3]
```diff
@@ -54,7 +56,7 @@ \subsection{The Core}
 \end{verbatim}
 
 Every SymPy expression satisfies a key invariant, namely,
-\texttt{expr.func(*expr.args) == expr}. This means that expressions are
+\texttt{expr.func(*expr.args) == expr}. This implies that expressions are
```
Comment (asmeurer) [inline architecture.tex]: I think "means" is better here.

File: architecture.tex [hunk 4]
```diff
@@ -244,5 +247,4 @@ \subsection{Extensibility}
             raise ArgumentIndexError(self, argindex)
 \end{verbatim}
 
-The actual gamma function defined in SymPy has much more implemented than
-this, such as evaluation at rational points and series expansion.
+The actual gamma function defined in SymPy, \texttt{scipy.stats.gamma}, has many more capabilities, such as evaluation at rational points and series expansion.
```
Comment (asmeurer) [inline architecture.tex]: This has nothing to do with scipy

**Comment by aktech** [comment]
Hey @fabianp nice to see you. :)

**Comment by fabianp** [comment]
Thanks for your comments @asmeurer . I took them into account. 

Also, let me know where in particular I could help. The parts in which I was involved -- the logic and assumption system-- seem to be already covered, but maybe I could work on comparing in with other systems (Maple? Maxima?)

**Comment by asmeurer** [comment]
I pushed a fix to this at https://github.com/sympy/sympy-paper/pull/92. Will merge when tests pass.

**Pull request closed without merging**